File: Aufgabe1/solutions/1.4/sqrt.cpp
#include <iostream>

#include "sqrt_lib.h"

int main() {
  double x = 0.0;
  std::cout << "Bitte geben Sie eine Zahl ein: ";
  std::cin >> x;
 
  std::cout << "Die Wurzel aus " << x << " ist " << sqrt_newton(x) << std::endl;
 
  return 0;
}

File: Aufgabe1/solutions/1.4/sqrt.hs

import Sqrtlib
import System.IO

-- Program entry point
main :: IO ()
main = putStr "Bitte geben Sie eine Zahl ein: " >>
  hFlush stdout >>
  getLine >>= 
  (\xStr -> 
    let x = read xStr :: Double
    in putStr ("Die Wurzel aus " ++ xStr ++ " ist: ") >> putStrLn (show $ Sqrtlib.sqrt_newton' x))